awx_collection/plugins/module_utils/tower_awxkit.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from . tower_module import TowerModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from awxkit.api.client import Connection
+    from awxkit.api.pages.api import ApiV2
+    from awxkit.api import get_registered_page
+    HAS_AWX_KIT = True
+except ImportError:
+    HAS_AWX_KIT = False
+
+
+class TowerAWXKitModule(TowerModule):
+    connection = None
+    apiV2Ref = None
+
+    def __init__(self, argument_spec, **kwargs):
+        kwargs['supports_check_mode'] = False
+
+        super(TowerAWXKitModule, self).__init__(argument_spec=argument_spec, **kwargs)
+
+        # Die if we don't have AWX_KIT installed
+        if not HAS_AWX_KIT:
+            self.exit_module(msg=missing_required_lib('awxkit'))
+
+        # Establish our conneciton object
+        self.connection = Connection(self.host, verify=self.verify_ssl)
+
+    def authenticate(self):
+        try:
+            self.connection.login(username=self.username, password=self.password, token=self.oauth_token)
+            # If we have neither of these, then we can try un-authenticated access
+            self.authenticated = True
```

Comment:
If you don't have username/pass or token, then wouldn't that error and put you in the `except` block?

Reply:
Those should all have a default of None from the TowerModule code:
```
    username = None
    password = None
    verify_ssl = True
    oauth_token = None
```

So they should never be "not defined" and cause an error.